plugin/contrib/mesos/pkg/executor/executor.go

```diff
@@ -231,6 +245,26 @@ func (k *KubernetesExecutor) onInitialRegistration() {
 	}
 }
 
+// InitializeStaticPodsSource blocks until initial regstration is complete and
+// then creates a static pod source using the given factory func.
+func (k *KubernetesExecutor) InitializeStaticPodsSource(sourceFactory func()) {
+	<-k.initialRegComplete
+
+	if k.staticPodsConfig == nil {
+		return
+	}
+
+	log.V(2).Infof("extracting static pods config to %s", k.staticPodsConfigPath)
+	err := archive.UnzipDirectory(k.staticPodsConfig, k.staticPodsConfigPath)
+	if err != nil {
+		log.Errorf("Failed to extract static pod config: %v", err)
+		return
+	}
+
+	log.V(2).Infof("initializing static pods source factory, configured at path %q", k.staticPodsConfigPath)
+	sourceFactory()
```

Comment:
What happens when the passed `sourceFactory` is nil or panics? I'm reticent in defaulting to callbacks for continuations and would prefer that a channel is used instead, decoupling execution and synchronisation.
In this case, we don't even need explicit sequencing with channels since `sourceFactory` is to be executed at the very end of the method.
```go
func (k *KubernetesExecutor) InitializeStaticPodsSource() error {
	<-k.initialRegComplete
	if k.staticPodsConfig == nil {
		return nil
	}
	log.V(2).Infof("extracting static pods config to %s", k.staticPodsConfigPath)
	return archive.UnzipDirectory(k.staticPodsConfig, k.staticPodsConfigPath)
}
```

```go
go func() {
    if err := executor.InitializeStaticPodsSource(); err != nil {
        log.Error(err)
        return
    }
    log.V(2).Infof("initializing static pods source factory, configured at path %q", staticPodsConfigPath)
    kconfig.NewSourceFile(staticPodsConfigPath, hostname, 1*time.Second, updates)
}()
```